### PR TITLE
Adds job-specific hazard vests!

### PR DESCRIPTION
**Pull request by Sadhorizon**
## What Does This PR Do
Adds a hazard vest specifically tailored for:
- Chief Engineer - green vest with white stripe. Has minor armor against radiation and fire. Can hold a telebaton and engi items. Can be found in CE's garment bag.
- Engineers - yellow vest with orange+gray stripe. Can hold engi items. Can be found on station, in engineering supply crate, in engineer lockers and in engidrobe.
- Atmos Techs - yellow vest with blue+gray stripe. Can hold engi items. Can be found on station, in atmos lockers and in atmosdrobe.
- Cargo Technicians - orange vest with yellow stripe. Can hold cargo items. Can be found in cargodrobe.
- Quartermaster - orange+black vest with yellow stripe. Can hold telebaton (in preparation for #22133) and cargo items. Can be found in QM's garment bag.
- Security - red vest with white stripe and a printed badge at the back. Has minor brute, laser and bomb armor (you are getting two more pockets, that's the tradeoff. Less armor.). Can hold security items. Can be found in secdrobe.
- Paramedic - blue+orange vest with white stripes. Has minor armor against rads and fire. Can hold medical items. Can be found in paramed's locker.
- General population - orange vest with white stripe. Can hold basic items. Can be found on station, in autodrobe, in janidrobe and in lawdrobe (IAAs go to hazardous areas too!)

This PR also adds Hazard Vest sprites for Kida!

## Why It's Good For The Game
One thing I love about labcoats is their shoulder stripes. You look at a nerd, see green stripes, and can immediately say "Yup, that's viro.". This PR adds that functionality to hazard vests too.

## Images of changes
PR

## Testing
Spawned in, took above screenshot, changed species, added stuff to the suit slot.

## Changelog
:cl:
add: Engineering, Atmos, CE, Cargo, QM, Security and Paramedic now have unique hazard vests.
add: Adds job-unique hazard vests to right vendors.
tweak: Basic hazard vest can now be bought in Autodrobes, Janidrobes and Lawdrobes.
add: QM and CE now have their hazard vests in their garment bags.
tweak: Atmos lockers now start with atmos vests, engi lockers now start with engi vests, paramed's locker now starts with paramed's vest.
tweak: "Engineering Gear Crate" now contains engineering hazard vests instead of basic ones.
imageadd: Kida now have unique hazard vest sprites.
/:cl:

### CODE REVIEW

**Comment by ArcticValor** [comment]
Space OSHA approved!

**Comment by Sadhorizon** [comment]
Updated sprites now, and added paramedic's vest

File: code/game/machinery/vendors/generic_vendors.dm [hunk 1]
```diff
@@ -509,6 +509,7 @@
 					/obj/item/clothing/mask/gas/plaguedoctor = 1,
 					/obj/item/clothing/suit/apron = 1,
 					/obj/item/clothing/under/misc/waiter = 1,
+					/obj/item/clothing/suit/storage/hazardvest = 2,
```
Comment (Henri215) [inline code/game/machinery/vendors/generic_vendors.dm:512]: Autodrobe is mostly for costumes, i would put it in another vendor.

File: code/modules/clothing/suits/hazard_vest.dm [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	name = "quartermaster's hazard vest"
+	desc = "A high-visibility vest used in work zones. This one belongs to quartermaster."
+	icon_state = "hazard_QM"
+	item_state = "hazard_QM"
+	allowed = list (/obj/item/flashlight, /obj/item/paper, /obj/item/clipboard, /obj/item/tank/internals/emergency_oxygen, /obj/item/rcs, /obj/item/destTagger, /obj/item/analyzer, /obj/item/wrench, /obj/item/radio, /obj/item/melee/classic_baton/telescopic) // Same as regular, but also can hold their telebaton
+
+/*SECURITY*/
+/obj/item/clothing/suit/storage/hazardvest/sec
+	name = "security hazard vest"
+	desc = "A high-visibility vest used in work zones. This one has security colors. It does not look durable."
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:56]: well.. its more durable than the others :D

File: code/modules/clothing/suits/hazard_vest.dm [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	icon_state = "hazard_sec"
+	item_state = "hazard_sec"
+	allowed = list (/obj/item/clipboard, /obj/item/analyzer, /obj/item/tank/internals/emergency_oxygen, /obj/item/gun/energy, /obj/item/reagent_containers/spray/pepper, /obj/item/gun/projectile, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/restraints/handcuffs, /obj/item/flashlight, /obj/item/melee/classic_baton/telescopic, /obj/item/radio, /obj/item/clothing/accessory/holobadge)
+	armor = list(MELEE = 5, BULLET = 5, LASER = 5, ENERGY = 0, BOMB = 5, RAD = 0, FIRE = 20, ACID = 20) // Weak armor, but you also get two pockets. Why are you wearing a cheap piece of fabric for combat anyways?
+
+/*MEDICAL*/
+/obj/item/clothing/suit/storage/hazardvest/paramed
+	name = "paramedic's hazard vest"
+	desc = "A high-visibility vest used in work zones. This one belongs to paramedic."
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:65]: Could add a description here like the CE one, since they share the same armor values.

**Review by Henri215 — COMMENTED** [review]
Lets hope one day hazard vests sprites are split into their own dmi file since they got a good amount now :(

File: code/modules/clothing/suits/hazard_vest.dm (hunk 2 above)
Comment (Sadhorizon) [reply code/modules/clothing/suits/hazard_vest.dm:56]: it's less durable than the armor. I want to make it clear.

**Comment by Sadhorizon** [comment]
> Lets hope one day hazard vests sprites are split into their own dmi file since they got a good amount now :(

Currently, ALL of suit sprites are in suit.dmi
Only jumpsuits were split so far.

File: code/modules/clothing/suits/hazard_vest.dm [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	item_state = "hazard_QM"
+	allowed = list (/obj/item/flashlight, /obj/item/paper, /obj/item/clipboard, /obj/item/tank/internals/emergency_oxygen, /obj/item/rcs, /obj/item/destTagger, /obj/item/analyzer, /obj/item/wrench, /obj/item/radio, /obj/item/melee/classic_baton/telescopic) // Same as regular, but also can hold their telebaton
+
+/*SECURITY*/
+/obj/item/clothing/suit/storage/hazardvest/sec
+	name = "security hazard vest"
+	desc = "A high-visibility vest used in work zones. This one has security colors. It does not look durable."
+	icon_state = "hazard_sec"
+	item_state = "hazard_sec"
+	allowed = list (/obj/item/clipboard, /obj/item/analyzer, /obj/item/tank/internals/emergency_oxygen, /obj/item/gun/energy, /obj/item/reagent_containers/spray/pepper, /obj/item/gun/projectile, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/restraints/handcuffs, /obj/item/flashlight, /obj/item/melee/classic_baton/telescopic, /obj/item/radio, /obj/item/clothing/accessory/holobadge)
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:59]:
```suggestion
	allowed = list(/obj/item/clipboard, /obj/item/analyzer, /obj/item/tank/internals/emergency_oxygen, /obj/item/gun/energy, /obj/item/reagent_containers/spray/pepper, /obj/item/gun/projectile, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/restraints/handcuffs, /obj/item/flashlight, /obj/item/melee/classic_baton/telescopic, /obj/item/radio, /obj/item/clothing/accessory/holobadge)
```

File: code/modules/clothing/suits/hazard_vest.dm [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	icon_state = "hazard_cargo"
+	item_state = "hazard_cargo"
+	allowed = list (/obj/item/flashlight, /obj/item/paper, /obj/item/clipboard, /obj/item/tank/internals/emergency_oxygen, /obj/item/rcs, /obj/item/destTagger, /obj/item/analyzer, /obj/item/wrench, /obj/item/radio)
+
+/obj/item/clothing/suit/storage/hazardvest/qm
+	name = "quartermaster's hazard vest"
+	desc = "A high-visibility vest used in work zones. This one belongs to quartermaster."
+	icon_state = "hazard_QM"
+	item_state = "hazard_QM"
+	allowed = list (/obj/item/flashlight, /obj/item/paper, /obj/item/clipboard, /obj/item/tank/internals/emergency_oxygen, /obj/item/rcs, /obj/item/destTagger, /obj/item/analyzer, /obj/item/wrench, /obj/item/radio, /obj/item/melee/classic_baton/telescopic) // Same as regular, but also can hold their telebaton
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:51]:
```suggestion
	allowed = list(/obj/item/flashlight, /obj/item/paper, /obj/item/clipboard, /obj/item/tank/internals/emergency_oxygen, /obj/item/rcs, /obj/item/destTagger, /obj/item/analyzer, /obj/item/wrench, /obj/item/radio, /obj/item/melee/classic_baton/telescopic) // Same as regular, but also can hold their telebaton
```

File: code/modules/clothing/suits/hazard_vest.dm [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, RAD = 5, FIRE = 50, ACID = 50) // CE is special enough that they can probably get some minor armor
+	allowed = list (/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd, /obj/item/melee/classic_baton/telescopic) // Same as regular, but also can hold their telebaton
+
+/*CARGO*/
+/obj/item/clothing/suit/storage/hazardvest/cargo
+	name = "cargo hazard vest"
+	desc = "A high-visibility vest used in work zones. This one has cargo colors."
+	icon_state = "hazard_cargo"
+	item_state = "hazard_cargo"
+	allowed = list (/obj/item/flashlight, /obj/item/paper, /obj/item/clipboard, /obj/item/tank/internals/emergency_oxygen, /obj/item/rcs, /obj/item/destTagger, /obj/item/analyzer, /obj/item/wrench, /obj/item/radio)
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:44]:
```suggestion
	allowed = list(/obj/item/flashlight, /obj/item/paper, /obj/item/clipboard, /obj/item/tank/internals/emergency_oxygen, /obj/item/rcs, /obj/item/destTagger, /obj/item/analyzer, /obj/item/wrench, /obj/item/radio)
```

File: code/modules/clothing/suits/hazard_vest.dm [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	item_state = "hazard_atmos"
+	allowed = list (/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd)
+
+/obj/item/clothing/suit/storage/hazardvest/ce
+	name = "chief engineer's hazard vest"
+	desc = "A high-visibility vest used in work zones. This one belongs to chief engineer. It's made out of strong material."
+	icon_state = "hazard_CE"
+	item_state = "hazard_CE"
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, RAD = 5, FIRE = 50, ACID = 50) // CE is special enough that they can probably get some minor armor
+	allowed = list (/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd, /obj/item/melee/classic_baton/telescopic) // Same as regular, but also can hold their telebaton
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:36]:
```suggestion
	allowed = list(/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd, /obj/item/melee/classic_baton/telescopic) // Same as regular, but also can hold their telebaton
```

File: code/modules/clothing/suits/hazard_vest.dm [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+	icon_state = "hazard_engi"
+	item_state = "hazard_engi"
+	allowed = list (/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd)
+
+/obj/item/clothing/suit/storage/hazardvest/atmos
+	name = "atmospheric hazard vest"
+	desc = "A high-visibility vest used in work zones. This one has atmospheric colors."
+	icon_state = "hazard_atmos"
+	item_state = "hazard_atmos"
+	allowed = list (/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd)
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:28]:
```suggestion
	allowed = list(/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd)
```

File: code/modules/clothing/suits/hazard_vest.dm [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+		"Kidan" = 'icons/mob/clothing/species/kidan/suit.dmi'
+		)
+
+/*ENGINEERING*/
+/obj/item/clothing/suit/storage/hazardvest/engi
+	name = "engineering hazard vest"
+	desc = "A high-visibility vest used in work zones. This one has engineering colors."
+	icon_state = "hazard_engi"
+	item_state = "hazard_engi"
+	allowed = list (/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd)
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:21]:
```suggestion
	allowed = list(/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/screwdriver, /obj/item/radio, /obj/item/wrench, /obj/item/weldingtool, /obj/item/crowbar, /obj/item/wirecutters, /obj/item/rcd, /obj/item/rpd)
```

File: code/modules/clothing/suits/hazard_vest.dm [hunk 10]
```diff
@@ -0,0 +1,69 @@
+/obj/item/clothing/suit/storage/hazardvest
+	name = "hazard vest"
+	desc = "A high-visibility vest used in work zones."
+	icon_state = "hazard"
+	item_state = "hazard"
+	blood_overlay_type = "armor"
+	allowed = list (/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/radio)
```
Comment (Henri215) [inline code/modules/clothing/suits/hazard_vest.dm:7]:
```suggestion
	allowed = list(/obj/item/flashlight, /obj/item/t_scanner, /obj/item/tank/internals/emergency_oxygen, /obj/item/clipboard, /obj/item/analyzer, /obj/item/radio)
```

**Review by Henri215 — APPROVED** [review]
Looks good to me after these changes are applied

**Comment by Sadhorizon** [comment]
McRamon requested some changes, I'll get to them later on.

**Comment by Burzah** [comment]
When you have a chance, please clear any conflict so this can get merged!

**Comment by Sadhorizon** [comment]
ah shit, I know what went wrong

**Review by Burzah — APPROVED** [review]

**Comment by Sadhorizon** [comment]
AAAAAAA
wrong branch

File: _maps/map_files/generic/Lavaland.dmm [hunk 11]
```diff
@@ -250,6 +250,10 @@
 	icon_state = "darkred"
 	},
 /area/mine/laborcamp/security)
+"aL" = (
```
Comment (farie82) [inline _maps/map_files/generic/Lavaland.dmm:253]: Are these changes part of the PR?

Comment (Sadhorizon) [reply _maps/map_files/generic/Lavaland.dmm:253]: NOT AT ALL

**Review by hal9000PR — CHANGES_REQUESTED** [review]
please remove the map changes and deconflict for merge

**Comment by Sadhorizon** [comment]
I have lost motivation to work on this anymore. Feel free to take the sprites and re-open.

**Pull request closed without merging**